### DISCUSSION TITLE
How does a streaming response work? When does it start and end? How can I check?

### DISCUSSION THREAD

**Discussion by Chris230291** [fastapi/fastapi#8826] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
ffmpegcmd = [
	'ffmpeg',
	'-re',
	'-loglevel', 'panic', '-hide_banner',
	'-i', *url*,
	'-vcodec', 'copy',
	'-f', 'mp4',
	'-movflags', 'frag_keyframe+empty_moov',
	'pipe:'
]

@app.get('/stream')
def stream():
    def streamData():
        with subprocess.Popen(ffmpegcmd, stdin=subprocess.DEVNULL, stdout=subprocess.PIPE) as ffmpeg_sb:
            yield from iter(ffmpeg_sb.stdout.readline, "")
    return StreamingResponse(streamData(), media_type="video/mp4")
```

### Description

Suppose I have the code I posted above. When I connect, ffmpeg starts restreaming the source stream and it plays. If I close my web browser or media player the ffmpeg process ends. This is all good. But what happens if the source stream that ffmpeg uses glitches or goes down and ffmpeg exits by itself? Would fastapi close the connection to the client? I ask this because I would expect a player like vlc (with repeat enabled) to simply make a fresh request to fastapi when it stops receiving the video and start the process all over again, but it doesn't. Is this because fastapi is still "streaming" and keeping the connection alive somehow? I'd like to learn what is happening so I can attempt to find the cause and fix this issue.

Thanks,
Chris

### Operating System

Windows

### Operating System Details

Windows 10

### FastAPI Version

0.68.0

### Python Version

Python 3.8.3

**Comment by ghandic** [comment]
What would the steps be to reproduce the issue? I'm assuming you'll have to force an issue with ffmpeg and then handle that error in the stream to retry

**Comment by Chris230291** [comment]
Hello. I can simply close the ffmpeg process in windows task manager to simulate ffmpeg exiting.
